### PR TITLE
separate concerns into evaluator and nullator

### PR DESCRIPTION
**Pull request by havvg**
@Nek- what do you think?

This would be a full BC break by making this `v2.0.0`.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -55,13 +61,29 @@ services:
...
+    tarifhaus.doctrine.nullable_embeddable_listener:
+        public: false
+        class: Tarifhaus\Doctrine\ORM\NullableEmbeddableListener
+        arguments:
+            - '@property_accessor'
+            - '@tarifhaus.doctrine.nullable_embeddable_closure_nullator'
+        calls:
+            - ['addMapping', ['Thengine\Domain\User\Model\UserProfile', 'address']]
+        tags:
+            - { name: 'doctrine.orm.entity_listener', entity: '\Thengine\Domain\User\Model\UserProfile', event: 'postLoad' }
```
Comment (Nek-) [inline README.md]: Adding configuration for Symfony4 would be nice :3 (implies many lines)

File: README.md [hunk 2]
```diff
@@ -55,13 +61,29 @@ services:
...
             - '@property_accessor'
+            - '@property_accessor'
         calls:
             - ['addMapping', ['Thengine\Domain\User\Model\UserProfile', 'address']]
         tags:
             - { name: 'doctrine.orm.entity_listener', entity: '\Thengine\Domain\User\Model\UserProfile', event: 'postLoad' }
 ```
 
-# Usage
+There is also a `nullator` using a bound closure to set the `null` value privately.
```
Comment (Nek-) [inline README.md]: This should be the recommended way, don't you think so?

**Review by Nek- — APPROVED** [review]
This looks great !

File: README.md (hunk 2 above)
Comment (havvg) [reply README.md]: Yes, you are right. I will rewrite the section accordingly.

File: README.md (hunk 1 above)
Comment (havvg) [reply README.md]: What exactly do you mean? What's missing?

**Pull request merged**